marshal.go

```diff
@@ -10,6 +10,13 @@ type Enc struct {
 	off *uint64
 }
 
+func MakeEncFromSlice(b []byte) Enc {
```

Comment:
I think I'd prefer to call this `NewEncFromSlice`, to be consistent with the other constructor.